### ISSUE TITLE
Delete child schema

### ISSUE THREAD

**Issue by lobor** [materik/restberry#10]
It's possible, when call on delete /api/sci/:id, delete to /api/sci/:id/biens ?
Or i should delete biens manually with delete /api/biens/:id ?

if it's second solution, it's uncomfortable.

**Comment by materik** [comment]
There is a hook you can use for this. Define ``preRemove`` for your ``sci`` model and you can delete the biens belonging to the instance:
```
restberry.model('Sci')
    .preRemove(function(next) {
        var Bien = restberry.model('Bien');
        Bien.remove({sci: this.id}, next);
    })
```
Or something like that.

**Comment by lobor** [comment]
Sorry but the preRemove doesn't work. the preSave works.

**Comment by materik** [comment]
How does it not work?

**Comment by lobor** [comment]
I use it like this
```javascript
restberry
    .model('Sci')
    .preRemove(function(next){
        console.log('toto');
    });
```

And there are not a console log

**Comment by materik** [comment]
hmm. do you do a deleteAPI call or how do you delete the object?

**Comment by lobor** [comment]
```javascript
restberry
    .model('Sci')
    .schema({
        name: {
    		type: String,
    		required: true,
    		index: {
    			unique: true
    		}
    	},
        rcs: {
    		type: String,
    		required: true
    	},
        date_immatriculation: {
    		type: Date,
    		required: true
    	},
        capital: {
    		type: Number,
    		required: true
    	},
        naf: {
    		type: String,
    		required: true
    	},
        adress: {
    		type: String,
    		required: true
    	},
        cp: {
    		type: Number,
    		required: true
    	},
        city: {
    		type: String,
    		required: true
    	},
        nb_part: {
    		type: Number,
    		required: true
    	},
        associes:[{
            nom: {
        		type: String,
        		required: true,
        	},
            prenom: {
        		type: String,
        		required: true
        	},
            address: {
        		type: String,
        		required: true
        	},
            cp : {
                type: Number,
                required: true
            },
            city: {
                type: String,
                required: true
            },
            birthday: {
                type: Date,
                required: true
            },
            city_birthday: {
                type: String,
                required: true
            },
            np_part: {
                type: Number,
                required: true
            },
            percent_part: {
                type: Number,
                required: true
            }
        }]
    })
    .preRemove(function(next) {
        var Bien = restberry.model('Bien');
        Bien.remove({sci: this.id}, next);
    })
    .loginRequired()
    .routes
    .addCRUDRoutes();
```

It's delete api rest delete /api/sci/:id

**Comment by materik** [comment]
ok. And the Bien objects aren't removed I take it?

**Comment by lobor** [comment]
yes, it seems to me it's not executed

**Comment by materik** [comment]
will take a look this evening. seems strange since i do have test for it. you could check the tests out yourself, otherwise I will get back to you later.

**Comment by materik** [comment]
Awesome catch @lobor!!! I was able to add a test which reproduced your problems. It seems like the preRemove function aren't called. I'll fix this right away!

**Comment by materik** [comment]
![screen shot 2015-06-26 at 11 24 52 pm](https://cloud.githubusercontent.com/assets/2392010/8387647/964c9a34-1c5a-11e5-9f76-ff05f4a3f46f.png)
Easiest fix ever!! Just called the wrong function :)

**Comment by materik** [comment]
So, now it's fixed in 1.1.1
